File: Python/BinaryTreeLevelOrderTraversal.py
from collections import deque

class TreeNode:
    def __init__(self,val=0,left=None,right=None) -> None:
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def levelOrderTraversal(root):

        if root is None:
            return []
        
        queue = deque()
        outputArray = []

        while len(queue)> 0 :
            innerArray = []
            for i in range(len(queue)):
                cur = queue.popleft()
                innerArray.append(cur.val)
                if cur.left:
                    queue.append(cur.left)
                if cur.right:
                    queue.append(cur.right)

            outputArray.append(innerArray)

        return outputArray

File: Javascript/BinaryTreeLevelOrderTraversal.js
function TreeNode(val,left,right){
    this.val = (val === undefined) ? 0:val 
    this.left = (left === undefined) ? null:left 
    this.right = (right === undefined) ? null:right 
}

var levelOrderTraversal = function(root){
    if(root === null){
        return []
    }

    const queue = []
    const outputArray = []

    queue.push(root)

    while(queue.length>0){
        const innerArray = []
        let queueLength = queue.length
        for(let i=0;i<queueLength;i++){
            const cur = queue.shift()
            innerArray.push(cur.val)
            if(cur.left !== null){
                queue.push(cur.left)
            }
            if(cur.right !== null){
                queue.push(cur.right)
            }
        }
        outputArray.push(innerArray)
    }

    return outputArray
}